### ISSUE TITLE
Possibly false positives if library doesn't depend on libc

### ISSUE THREAD

**Issue by mattiaverga** [rpm-software-management/rpmlint#786]
For a couple of packages I'm getting what I think are false positives from rpmlint.
In [these libraries](https://copr.fedorainfracloud.org/coprs/mattia/Astronomy/build/3136089/) I get:
```
indi-3rdparty-libapogee.x86_64: E: shared-library-without-dependency-information /usr/lib64/libapogee.so.3.2
indi-3rdparty-libapogee.x86_64: W: no-soname /usr/lib64/libapogee.so.3.2
indi-3rdparty-libapogee.x86_64: E: missing-PT_GNU_STACK-section /usr/lib64/libapogee.so.3.2
```
while [these drivers](https://copr.fedorainfracloud.org/coprs/mattia/Astronomy/build/3136090/) reports:
```
indi-3rdparty-aagcloudwatcher-ng.x86_64: E: statically-linked-binary /usr/bin/aagcloudwatcher_test_ng
indi-3rdparty-aagcloudwatcher-ng.x86_64: E: statically-linked-binary /usr/bin/indi_aagcloudwatcher_ng
indi-3rdparty-aagcloudwatcher-ng.x86_64: E: missing-PT_GNU_STACK-section /usr/bin/aagcloudwatcher_test_ng
indi-3rdparty-aagcloudwatcher-ng.x86_64: E: missing-PT_GNU_STACK-section /usr/bin/indi_aagcloudwatcher_ng
```

I'm no expert, but running ldd on the produced library I think it's dynamically linked, not statically:
```
$ ldd usr/lib64/libapogee.so.3.2
        linux-vdso.so.1 (0x00007fff4cf81000)
        libusb-1.0.so.0 => /lib64/libusb-1.0.so.0 (0x00007f6af2724000)
        libcurl.so.4 => /lib64/libcurl.so.4 (0x00007f6af267f000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f6af2460000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f6af2445000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f6af223b000)
        libudev.so.1 => /lib64/libudev.so.1 (0x00007f6af220f000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f6af2827000)
        libnghttp2.so.14 => /lib64/libnghttp2.so.14 (0x00007f6af21e5000)
        libidn2.so.0 => /lib64/libidn2.so.0 (0x00007f6af21c4000)
        libssh.so.4 => /lib64/libssh.so.4 (0x00007f6af2155000)
        libpsl.so.5 => /lib64/libpsl.so.5 (0x00007f6af2141000)
        libssl.so.1.1 => /lib64/libssl.so.1.1 (0x00007f6af20a4000)
        libcrypto.so.1.1 => /lib64/libcrypto.so.1.1 (0x00007f6af1db6000)
        libgssapi_krb5.so.2 => /lib64/libgssapi_krb5.so.2 (0x00007f6af1d5d000)
        libkrb5.so.3 => /lib64/libkrb5.so.3 (0x00007f6af1c7f000)
        libk5crypto.so.3 => /lib64/libk5crypto.so.3 (0x00007f6af1c67000)
        libcom_err.so.2 => /lib64/libcom_err.so.2 (0x00007f6af1c60000)
        libldap_r-2.4.so.2 => /lib64/libldap_r-2.4.so.2 (0x00007f6af1c04000)
        liblber-2.4.so.2 => /lib64/liblber-2.4.so.2 (0x00007f6af1bf1000)
        libbrotlidec.so.1 => /lib64/libbrotlidec.so.1 (0x00007f6af1be1000)
        libz.so.1 => /lib64/libz.so.1 (0x00007f6af1bc7000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f6af1aeb000)
        libunistring.so.2 => /lib64/libunistring.so.2 (0x00007f6af1966000)
        libkrb5support.so.0 => /lib64/libkrb5support.so.0 (0x00007f6af1955000)
        libkeyutils.so.1 => /lib64/libkeyutils.so.1 (0x00007f6af194c000)
        libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f6af1938000)
        libsasl2.so.3 => /lib64/libsasl2.so.3 (0x00007f6af1918000)
        libbrotlicommon.so.1 => /lib64/libbrotlicommon.so.1 (0x00007f6af18f5000)
        libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f6af18ca000)
        libcrypt.so.2 => /lib64/libcrypt.so.2 (0x00007f6af188e000)
        libpcre2-8.so.0 => /lib64/libpcre2-8.so.0 (0x00007f6af17f7000)
```
and
```
$ ldd usr/bin/aagcloudwatcher_test_ng 
        linux-vdso.so.1 (0x00007ffcdb5fc000)
        libindidriver.so.1 => /lib64/libindidriver.so.1 (0x00007fcd9797e000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007fcd9775f000)
        libm.so.6 => /lib64/libm.so.6 (0x00007fcd97683000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007fcd97668000)
        libc.so.6 => /lib64/libc.so.6 (0x00007fcd9745e000)
        libusb-1.0.so.0 => /lib64/libusb-1.0.so.0 (0x00007fcd97441000)
        libnova-0.15.so.0 => /lib64/libnova-0.15.so.0 (0x00007fcd970be000)
        libcfitsio.so.9 => /lib64/libcfitsio.so.9 (0x00007fcd96da8000)
        libz.so.1 => /lib64/libz.so.1 (0x00007fcd96d8e000)
        libjpeg.so.62 => /lib64/libjpeg.so.62 (0x00007fcd96d0c000)
        libfftw3.so.3 => /lib64/libfftw3.so.3 (0x00007fcd96b1a000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fcd97b7e000)
        libudev.so.1 => /lib64/libudev.so.1 (0x00007fcd96aee000)
        libbz2.so.1 => /lib64/libbz2.so.1 (0x00007fcd96ad9000)
```

Also, the soname is there:
```
$ objdump -p usr/lib64/libapogee.so.3.2 | grep SONAME
  SONAME               libapogee.so.3
```

As suggested in [this discussion](https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/EURXRSOAJZOIGJRIFRRPAOXXZTX3FTT7/) on Fedora devel mailing list, I made sure to pass `-Wa,--noexecstack` to the build flags and `-Wl,-z,noexecstack` to the linker flags. There it is claimed that all libraries that do not depend on any symbols in libc are showing these errors.

**Comment by marxin** [comment]
> For a couple of packages I'm getting what I think are false positives from rpmlint. In [these libraries](https://copr.fedorainfracloud.org/coprs/mattia/Astronomy/build/3136089/) I get:
> 
> ```
> indi-3rdparty-libapogee.x86_64: E: shared-library-without-dependency-information /usr/lib64/libapogee.so.3.2
> indi-3rdparty-libapogee.x86_64: W: no-soname /usr/lib64/libapogee.so.3.2
> indi-3rdparty-libapogee.x86_64: E: missing-PT_GNU_STACK-section /usr/lib64/libapogee.so.3.2
> ```

Hello. I can't reproduce any of these errors.

> 
> while [these drivers](https://copr.fedorainfracloud.org/coprs/mattia/Astronomy/build/3136090/) reports:
> 
> ```
> indi-3rdparty-aagcloudwatcher-ng.x86_64: E: statically-linked-binary /usr/bin/aagcloudwatcher_test_ng
> indi-3rdparty-aagcloudwatcher-ng.x86_64: E: statically-linked-binary /usr/bin/indi_aagcloudwatcher_ng
> indi-3rdparty-aagcloudwatcher-ng.x86_64: E: missing-PT_GNU_STACK-section /usr/bin/aagcloudwatcher_test_ng
> indi-3rdparty-aagcloudwatcher-ng.x86_64: E: missing-PT_GNU_STACK-section /usr/bin/indi_aagcloudwatcher_ng
> ```

And similarly here.

What rpmlint version do you use? What configuration do you use? Please copy here output with `--verbose` argument.

**Comment by mattiaverga** [comment]
Hello,
I'm using rpmlint 2.2.0 under Fedora 35.
I'll attach the full output as files, because for libapogee the output is long.

[aagcloudwatcher.txt](https://github.com/rpm-software-management/rpmlint/files/7839388/aagcloudwatcher.txt)
[libapogee.txt](https://github.com/rpm-software-management/rpmlint/files/7839389/libapogee.txt)

**Comment by marxin** [comment]
And please provide direct links where I can download the rpm files.

**Comment by mattiaverga** [comment]
```
$ readelf -Wd ./usr/lib64/libapogee.so.3.2

Dynamic section at offset 0xc0568 contains 31 entries:
  Tag        Tipo                         Nome/Valore
 0x0000000000000001 (NEEDED)             Libreria condivisa: [libusb-1.0.so.0]
 0x0000000000000001 (NEEDED)             Libreria condivisa: [libcurl.so.4]
 0x0000000000000001 (NEEDED)             Libreria condivisa: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Libreria condivisa: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Libreria condivisa: [libc.so.6]
 0x000000000000000e (SONAME)             soname della libreria: [libapogee.so.3]
 0x000000000000000c (INIT)               0x26000
 0x000000000000000d (FINI)               0x9d9f0
 0x0000000000000019 (INIT_ARRAY)         0xbf328
 0x000000000000001b (INIT_ARRAYSZ)       16 (byte)
 0x000000000000001a (FINI_ARRAY)         0xbf338
 0x000000000000001c (FINI_ARRAYSZ)       8 (byte)
 0x000000006ffffef5 (GNU_HASH)           0x310
 0x0000000000000005 (STRTAB)             0xb380
 0x0000000000000006 (SYMTAB)             0x2b60
 0x000000000000000a (STRSZ)              60451 (byte)
 0x000000000000000b (SYMENT)             24 (byte)
 0x0000000000000003 (PLTGOT)             0xc1798
 0x0000000000000002 (PLTRELSZ)           14952 (byte)
 0x0000000000000014 (PLTREL)             RELA
 0x0000000000000017 (JMPREL)             0x22290
 0x0000000000000007 (RELA)               0x1ac40
 0x0000000000000008 (RELASZ)             30288 (byte)
 0x0000000000000009 (RELAENT)            24 (byte)
 0x000000000000001e (FLAGS)              BIND_NOW
 0x000000006ffffffb (FLAGS_1)            Flag: NOW NODUMP
 0x000000006ffffffe (VERNEED)            0x1ab00
 0x000000006fffffff (VERNEEDNUM)         3
 0x000000006ffffff0 (VERSYM)             0x19fa4
 0x000000006ffffff9 (RELACOUNT)          308
 0x0000000000000000 (NULL)               0x0
```

**Comment by mattiaverga** [comment]
My guess: the parser only works with English locale? I'm using Italian locale.

**Comment by marxin** [comment]
That looks correct, can you please checkout this repository and run `./lint.py -c configs/Fedora ...`, do you see the same error?

**Comment by mattiaverga** [comment]
Yes, same output with `./lint.py ...`

May the locale used be the problem? I see the regex used in the parser expects English output, while I'm on Italian locale. I did try to override the locale with `export LC_ALL=C`, but I get always the same output.

**Comment by marxin** [comment]
> May the locale used be the problem? I see the regex used in the parser expects English output, while I'm on Italian locale. I did try to override the locale with `export LC_ALL=C`, but I get always the same output.

That should be fine as we use english locale:
https://github.com/rpm-software-management/rpmlint/blob/df06abf975c140115c2fbcbcb71800997195759b/rpmlint/readelfparser.py#L88-L89

Can you please show me what you get with:
```patch
diff --git a/rpmlint/checks/BinariesCheck.py b/rpmlint/checks/BinariesCheck.py
index 4bda8563..56bf16e3 100644
--- a/rpmlint/checks/BinariesCheck.py
+++ b/rpmlint/checks/BinariesCheck.py
@@ -433,6 +433,10 @@ class BinariesCheck(AbstractCheck):
             return
 
         dyn_section = self.readelf_parser.dynamic_section_info
+        print(dyn_section)
+        print(dyn_section.needed)
+        print(dyn_section.soname)
+        print(self.is_shobj)
         if not len(dyn_section.needed) and not (dyn_section.soname and
                                                 self.ldso_soname_regex.search(dyn_section.soname)):
             if self.is_shobj:
```

**Comment by mattiaverga** [comment]
```
$ ./lint.py -c configs/Fedora ../indi-3rdparty-libapogee-1.9.3-3.fc36.x86_64.rpm 
<rpmlint.readelfparser.ElfDynamicSectionInfo object at 0x7f125d160f40>
[]
None
True
================================= rpmlint session starts ================================
rpmlint: 2.2.0
configuration:
    /home/mattia/rpm/rpmlint/rpmlint/configdefaults.toml
    /etc/xdg/rpmlint/fedora.toml
    /etc/xdg/rpmlint/licenses.toml
    /etc/xdg/rpmlint/scoring.toml
    /etc/xdg/rpmlint/users-groups.toml
    /etc/xdg/rpmlint/warn-on-functions.toml
    configs/Fedora/fedora.toml
    configs/Fedora/licenses.toml
    configs/Fedora/scoring.toml
    configs/Fedora/users-groups.toml
    configs/Fedora/warn-on-functions.toml
checks: 32, packages: 1

indi-3rdparty-libapogee.x86_64: E: shared-library-without-dependency-information /usr/lib64/libapogee.so.3.2
indi-3rdparty-libapogee.x86_64: W: no-soname /usr/lib64/libapogee.so.3.2
indi-3rdparty-libapogee.x86_64: E: missing-PT_GNU_STACK-section /usr/lib64/libapogee.so.3.2
indi-3rdparty-libapogee.x86_64: E: incorrect-fsf-address /usr/share/licenses/indi-3rdparty-libapogee/LICENSE
...
```

I also did:
```
$ pipenv shell
$ python
Python 3.10.1 (main, Dec  9 2021, 00:00:00) [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from rpmlint.readelfparser import ElfDynamicSectionInfo
>>> p = ElfDynamicSectionInfo('../usr/lib64/libapogee.so.3.2', [])
>>> p['SONAME']
['soname della libreria: [libapogee.so.3]']
```
So, it doesn't use the English locale.
Even forcing LC_ALL I get the same, while `readelf` command uses the correct locale:
```
$ locale
LANG=it_IT.UTF-8
LC_CTYPE="it_IT.UTF-8"
LC_NUMERIC="it_IT.UTF-8"
LC_TIME="it_IT.UTF-8"
LC_COLLATE="it_IT.UTF-8"
LC_MONETARY="it_IT.UTF-8"
LC_MESSAGES="it_IT.UTF-8"
LC_PAPER="it_IT.UTF-8"
LC_NAME="it_IT.UTF-8"
LC_ADDRESS="it_IT.UTF-8"
LC_TELEPHONE="it_IT.UTF-8"
LC_MEASUREMENT="it_IT.UTF-8"
LC_IDENTIFICATION="it_IT.UTF-8"
LC_ALL=

$ export LC_ALL=C

$ locale
LANG=it_IT.UTF-8
LC_CTYPE="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_COLLATE="C"
LC_MONETARY="C"
LC_MESSAGES="C"
LC_PAPER="C"
LC_NAME="C"
LC_ADDRESS="C"
LC_TELEPHONE="C"
LC_MEASUREMENT="C"
LC_IDENTIFICATION="C"
LC_ALL=C

$ pipenv shell
$ readelf -Wd ../usr/lib64/libapogee.so.3.2

Dynamic section at offset 0xc0568 contains 31 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libusb-1.0.so.0]
 0x0000000000000001 (NEEDED)             Shared library: [libcurl.so.4]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x000000000000000e (SONAME)             Library soname: [libapogee.so.3]
 0x000000000000000c (INIT)               0x26000
 0x000000000000000d (FINI)               0x9d9f0
 0x0000000000000019 (INIT_ARRAY)         0xbf328
 0x000000000000001b (INIT_ARRAYSZ)       16 (bytes)
 0x000000000000001a (FINI_ARRAY)         0xbf338
 0x000000000000001c (FINI_ARRAYSZ)       8 (bytes)
 0x000000006ffffef5 (GNU_HASH)           0x310
 0x0000000000000005 (STRTAB)             0xb380
 0x0000000000000006 (SYMTAB)             0x2b60
 0x000000000000000a (STRSZ)              60451 (bytes)
 0x000000000000000b (SYMENT)             24 (bytes)
 0x0000000000000003 (PLTGOT)             0xc1798
 0x0000000000000002 (PLTRELSZ)           14952 (bytes)
 0x0000000000000014 (PLTREL)             RELA
 0x0000000000000017 (JMPREL)             0x22290
 0x0000000000000007 (RELA)               0x1ac40
 0x0000000000000008 (RELASZ)             30288 (bytes)
 0x0000000000000009 (RELAENT)            24 (bytes)
 0x000000000000001e (FLAGS)              BIND_NOW
 0x000000006ffffffb (FLAGS_1)            Flags: NOW NODUMP
 0x000000006ffffffe (VERNEED)            0x1ab00
 0x000000006fffffff (VERNEEDNUM)         3
 0x000000006ffffff0 (VERSYM)             0x19fa4
 0x000000006ffffff9 (RELACOUNT)          308
 0x0000000000000000 (NULL)               0x0

$ python
Python 3.9.9 (main, Nov 19 2021, 00:00:00)
[GCC 11.2.1 20210728 (Red Hat 11.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import locale
>>> locale.getlocale()
(None, None)
>>> from rpmlint.readelfparser import ElfDynamicSectionInfo
>>> p = ElfDynamicSectionInfo('../usr/lib64/libapogee.so.3.2', [])
>>> p['SONAME']
['soname della libreria: [libapogee.so.3]']
>>> p.soname
>>>
```

**Comment by marxin** [comment]
Hmm, that's quite interesting. So we can play with the following demo:

```python#!/usr/bin/env python3

import os
import subprocess

ENGLISH_ENVIROMENT = dict(os.environ, LC_ALL='en_US.UTF-8')

r = subprocess.check_output('readelf -Wd /lib64/libc.so.6', shell=True, encoding='utf8')
lines = r.splitlines()
print([l for l in lines if 'SONAME' in l])

```

**Comment by marxin** [comment]
Can you please also test `ENGLISH_ENVIROMENT = dict(os.environ, LC_ALL='en_US.UTF-8', LANG='en_US.UTF-8')`?

**Comment by marxin** [comment]
And please test also:
```
import locale
locale.setlocale(locale.LC_ALL, 'en_US.UTF-8')
```

before subprocess call. Thanks.

**Comment by mattiaverga** [comment]
With Python 3.10:
```
$ python
Python 3.10.1 (main, Dec  9 2021, 00:00:00) [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> import subprocess
>>> 
>>> ENGLISH_ENVIROMENT = dict(os.environ, LC_ALL='en_US.UTF-8')
>>> 
>>> r = subprocess.check_output('readelf -Wd /lib64/libc.so.6', shell=True, env=ENGLISH_ENVIROMENT, encoding='utf8')
>>> lines = r.splitlines()
>>> print([l for l in lines if 'SONAME' in l])
[' 0x000000000000000e (SONAME)             soname della libreria: [libc.so.6]']
>>>
>>> ENGLISH_ENVIROMENT = dict(os.environ, LC_ALL='en_US.UTF-8', LANG='en_US.UTF-8')
>>> r = subprocess.check_output('readelf -Wd /lib64/libc.so.6', shell=True, env=ENGLISH_ENVIROMENT, encoding='utf8')
>>> lines = r.splitlines()>>> print([l for l in lines if 'SONAME' in l])[' 0x000000000000000e (SONAME)             soname della libreria: [libc.so.6]']
>>>
>>> import locale
>>> locale.setlocale(locale.LC_ALL, 'en_US.UTF-8')
'en_US.UTF-8'
>>> r = subprocess.check_output('readelf -Wd /lib64/libc.so.6', shell=True, env=ENGLISH_ENVIROMENT, encoding='utf8')
>>> lines = r.splitlines()
>>> print([l for l in lines if 'SONAME' in l])
[' 0x000000000000000e (SONAME)             soname della libreria: [libc.so.6]']
```
With Python 3.9:
```
$ python3.9
Python 3.9.9 (main, Nov 19 2021, 00:00:00) 
[GCC 11.2.1 20210728 (Red Hat 11.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> import subprocess
>>> 
>>> ENGLISH_ENVIROMENT = dict(os.environ, LC_ALL='en_US.UTF-8')
>>> 
>>> r = subprocess.check_output('readelf -Wd /lib64/libc.so.6', shell=True, env=ENGLISH_ENVIROMENT, encoding='utf8')
>>> lines = r.splitlines()
>>> print([l for l in lines if 'SONAME' in l])
[' 0x000000000000000e (SONAME)             soname della libreria: [libc.so.6]']
>>> 
>>> ENGLISH_ENVIROMENT = dict(os.environ, LC_ALL='en_US.UTF-8', LANG='en_US.UTF-8')
>>> r = subprocess.check_output('readelf -Wd /lib64/libc.so.6', shell=True, env=ENGLISH_ENVIROMENT, encoding='utf8')
>>> lines = r.splitlines()
>>> print([l for l in lines if 'SONAME' in l])
[' 0x000000000000000e (SONAME)             soname della libreria: [libc.so.6]']
>>> 
>>> import locale
>>> locale.setlocale(locale.LC_ALL, 'en_US.UTF-8')
'en_US.UTF-8'
>>> r = subprocess.check_output('readelf -Wd /lib64/libc.so.6', shell=True, env=ENGLISH_ENVIROMENT, encoding='utf8')
>>> lines = r.splitlines()
>>> print([l for l in lines if 'SONAME' in l])
[' 0x000000000000000e (SONAME)             soname della libreria: [libc.so.6]']
```

**Comment by mattiaverga** [comment]
mmm it looks like I get correct behavior if I run the script within a terminal, while under konsole (I'm running KDE on Wayland) can't correctly override the locale.

**Comment by mattiaverga** [comment]
So, definitely I think this is nothing to do with rpmlint.
The test script (and rpmlint) runs fine in a linux terminal both from my workstation and from my development virtual machine. The python environment cannot be overridden if I run the script from KDE Konsole (bot workstation and dev VM).

The strange is if I create a new user, login in KDE as the new user and run the script in Konsole, I get the correct output... so I think there's some setting in KDE/Plasma that makes python unable to change locale, but I haven't found yet what it is.
Thanks or your time.

**Comment by mattiaverga** [comment]
For reference, I found the culprit: in some circumstances, Konsole is setting the `LANGUAGE` environment variable and that was causing the issue. In my case that was set to `it:en_US` (which also seems odd).
Sorry for the noise.

**Comment by marxin** [comment]
Oh, I can reproduce that locally as well:

```python
LANGUAGE=it:en_US ./show.py
[' 0x000000000000000e (SONAME)             soname della libreria: [libc.so.6]']
```

So I think we should also set `LANGUAGE` when we call sub-processes, what do you think?

**Comment by mattiaverga** [comment]
Setting `LC_ALL` [has precedence](https://pubs.opengroup.org/onlinepubs/7908799/xbd/envvar.html) over all other LC_* and LANG, so it should not be needed to set also LANG:

> LC_ALL
>    This variable determines the values for all locale categories. The value of the LC_ALL environment variable has precedence over any of the other environment variables starting with LC_ (LC_COLLATE, LC_CTYPE, LC_MESSAGES, LC_MONETARY, LC_NUMERIC, LC_TIME) and the LANG environment variable. 

It doesn't mention LANGUAGE, though.

**Comment by marxin** [comment]
Thanks for the investigation, so let's updated only `LANGUAGE` then.
